pkg/cloudprovider/provider/packet/provider.go

```diff
@@ -0,0 +1,455 @@
+	instance, err := p.Get(machine)
+	if err != nil {
+		if err == cloudprovidererrors.ErrInstanceNotFound {
+			return true, nil
+		}
+		return false, err
+	}
+
+	c, _, _, err := p.getConfig(machine.Spec.ProviderSpec)
+	if err != nil {
+		return false, cloudprovidererrors.TerminalError{
+			Reason:  common.InvalidConfigurationMachineError,
+			Message: fmt.Sprintf("Failed to parse MachineSpec, due to %v", err),
+		}
+	}
+
+	client := getClient(c.APIKey)
+
+	res, err := client.Devices.Delete(instance.(*packetDevice).device.ID)
+	if err != nil {
+		return false, packetErrorToTerminalError(err, res, "failed to delete the server")
+	}
+
+	return false, nil
+}
+
+func (p *provider) AddDefaults(spec v1alpha1.MachineSpec) (v1alpha1.MachineSpec, error) {
+	return spec, nil
+}
+
+func (p *provider) Get(machine *v1alpha1.Machine) (instance.Instance, error) {
+	device, _, err := p.getPacketDevice(machine)
+	if err != nil {
+		return nil, err
+	}
+	if device != nil {
+		return &packetDevice{device: device}, nil
+	}
+
+	return nil, cloudprovidererrors.ErrInstanceNotFound
+}
+
+func (p *provider) MigrateUID(machine *v1alpha1.Machine, newID types.UID) error {
+	device, client, err := p.getPacketDevice(machine)
+	if err != nil {
+		return err
+	}
+
+	// go through existing labels, make sure that no other UID label exists
+	tags := make([]string, 0)
+	for _, t := range device.Tags {
+		// filter out old UID tag(s)
+		if _, err := getTagUID(t); err != nil {
+			tags = append(tags, t)
+		}
+	}
+
+	// create a new UID label
+	tags = append(tags, generateTag(string(newID)))
+
+	glog.Infof("Setting UID label for machine %s", machine.Name)
+	dur := &packngo.DeviceUpdateRequest{
+		Tags: &tags,
+	}
+	_, response, err := client.Devices.Update(device.ID, dur)
+	if err != nil {
+		return packetErrorToTerminalError(err, response, "failed to update UID label")
+	}
+	glog.Infof("Successfully set UID label for machine %s", machine.Name)
+
+	return nil
+}
+
+func (p *provider) GetCloudConfig(spec v1alpha1.MachineSpec) (config string, name string, err error) {
+	return "", "", nil
+}
+
+func (p *provider) MachineMetricsLabels(machine *v1alpha1.Machine) (map[string]string, error) {
+	labels := make(map[string]string)
+
+	c, _, _, err := p.getConfig(machine.Spec.ProviderSpec)
+	if err == nil {
+		labels["size"] = c.InstanceType
+		labels["facilities"] = strings.Join(c.Facilities, ",")
+	}
+
+	return labels, err
+}
+
+func (p *provider) SetMetricsForMachines(machines v1alpha1.MachineList) error {
+	return nil
+}
+
+type packetDevice struct {
+	device *packngo.Device
+}
+
+func (s *packetDevice) Name() string {
+	return s.device.Hostname
+}
+
+func (s *packetDevice) ID() string {
+	return s.device.ID
+}
+
+func (s *packetDevice) Addresses() []string {
+	// returns addresses in CIDR format
+	var addresses []string
+	for _, ip := range s.device.Network {
+		addresses = append(addresses, fmt.Sprintf("%s/%d", ip.Address, ip.CIDR))
```

Comment:
This must not contain the CIDR, just the address. For providers that don't have a cloud provider, we match the node by IP address and the nodes ip addresses dont have a CIDR in them

Reply:
Got it. Fixed.